src/apps/ojph_compress/ojph_compress.cpp

```diff
@@ -468,11 +468,9 @@ int main(int argc, char * argv[]) {
     "               lossy compression using the 9/7 wavelet transform;\n" 
     "               and true for reversible compression; that is,\n"
     "               lossless compression using the 5/3 wavelet transform.\n"
-    " -colour_trans (true) if there are three color components that are\n"
+    " -colour_trans (-1) if there are three color components that are\n"
```

Comment:
Hi Lucas,

Thank you for your continued check on the code.

-1 is for me to detect if the user has specified something.  The actual value is either true or false.  The default is true if you know that you have three color components that are known to be RGB.
The color transform converts RGB to YUV as specified in ITU-R BT.601; therefore, this transform does not make sense when the original file is already in YUV.  This transform improves coding efficiency.  

Perhaps, one can add that "This option is not available for yuv files."

By the way, the library should support rgba, but ojph_compress does not read these files yet.

Cheers,
Aous.

Reply:
Hi Lucas,

Sorry if I was confusing.  This is my thinking.

true if you want color transform, false if you do not want.

Color transform converts rgb to yuv.  Then, when image is yuv, we cannot employ color transform.

> ojph error 0x01000031 at ojph_compress.cpp:684: we currently do not support color transform on yuv files. In any case, this not a normal usage scenario. The OpenJPH library however does support that, but ojph_compress.cpp must be modified to send all lines from one component before moving to the next component; this requires buffering components outside of the OpenJPH library

I agree that this is confusing.  I think I should remove _The OpenJPH library however does support that, but ojph_compress.cpp must be modified to send all lines from one component before moving to the next component; this requires buffering components outside of the OpenJPH library_.  

Currently, I assume yuv is 1 or 3 color components.  We can easily (not tested) modify this to make it 4 components.  The Library itself should support a large number of components, but I never tested with more than 3.  Therefore, I may have bugs.  

I also assume that .yuv are in YUV or YCbCr color space, and therefore cannot employ color transform, because original is not in RGB.

I think I should have raw input files. In this case, all possible raw inputs can be handled including more than 3 components. Perhaps, I should leave it to the user to decide if (s)he wants color transform; not force anything on the user.

Let me know your thoughts.

Cheers,
Aous.

Reply:
_I also assume that .yuv are in YUV or YCbCr color space, and therefore cannot employ color transform, because original is not in RGB._
The joke right now is that no photo viewer I know can handle jpeg2000-yuv. Five years ago it was understandable because the world was 8bit and nobody wanted innovation. Now what to do with heif/avif 10<->12bit hdr images? Not popular, but it's for Windows. I read that jpeg 2000 part 16 is compatible with heif.

Reply:
> No photo viewer I know can handle jpeg2000-yuv444p

ffplay, mvp that is based on ffmpeg. Enough for me. ;)

Reply:
Just checked, Photoshop 2021 also supports it. Though it does not say it is YUV. Ha.

Also very funny is that exiftool thinks it is sYCC (it is not YUV and is not sRGB, it is extended gamut sRGB in YUV, where out-of-gamut, per sé negative RGB values are further defined with XYZ to linear RGB matrix).

Also, double checked Chrome does not support it too. Sigh.